### ISSUE TITLE
Ganache chews through storage space. Any way to combat this?

### ISSUE THREAD

**Issue by coreycaplan3** [trufflesuite/ganache-cli#678]
* I asked about this on [gitter](https://gitter.im/ConsenSys/truffle): [x]

* Link to question on [gitter](https://gitter.im/ConsenSys/truffle?at=5d7fdd313cc2b87edf1510cc)

## Current Behavior

I'm running ganache in a docker instance on AWS and it's currently chewing through storage space at an unbelievable pace (about 0.4Gb every hour). The chain currently receives a tx every 15 seconds (no more than 2 to 3 txs). Is there any way to prune the database or slow down the speed at which it grows?

## Expected Behavior

To not grow by 400 megabytes every hour.

## Possible Solution

N/A

## Steps to Reproduce (for bugs)

It should be as simple as creating your own chain and sending a steady stream of transactions to it.

## Your Environment
* Version used:
* Version of Truffle/Remix/Other tools used: 6.7.0 (ganache-cli) 2.8.0 (ganache-core)
* NodeJS Version: [ ] 6.x, [ ] 7.x (unsupported), [x] 8.x, [ ] 9.x
* Operating System and version (include distro if Linux):
* Link to your project or repro gist: N/A
* Commit hash to use with above link for reproduction: N/A
* I intend to submit a pull request to fix this issue: [ ]

**Comment by davidmurdoch** [comment]
Hey there @coreycaplan3! What are your startup options? Have you enabled interval mining?

**Comment by coreycaplan3** [comment]
Hey @davidmurdoch thanks for the reply! My startup options are:
```ganache-cli \
-i 1001 \
--quiet \
-d \
-k=petersburg \
-p 8545 \
-h 0.0.0.0 \
-a 40 \
-e 1000000 \
--mnemonic \"<my mnemonic phrase here>\" \
--db=/home/.ganache --gasLimit=0x7A1200```

I briefly tried using interval mining, but that cause my migrations to not be applied properly. Do you think that's a suitable fix?

**Comment by davidmurdoch** [comment]
No, interval mining would make it way worse, my hunch was that that was the problem :-(.

This is likely "normal" ganache db usage, unfortunately. I've been working on a fork (er, branch) of ganache-core that refactors our database use to use a binary data store instead of the mostly utf-8 store we use now. This work likely won't be completed for many more months though.

You could try to use ganache-core's `server` method (`require("ganache-core").server(options); ...`) and pass in your own database instance via the `db` option. Typically `memdown` is used here for its performance improvements, but you could try using `leveldown`, like this: `const server = require("ganache-core").server(leveldown(location)); server.listen(8545, () => { /* listening on 8545 */ });`. I haven't tested this, but it _should_ work. The resulting DB won't be compatible with ganache's internal database schema, but as long as you don't need to switch back to using ganache's internal db you'll be fine.

If this works for you please don't close this issue. I'd like to keep it Open to use to prioritize more performance-related work over the short term.
